lib/virt_autotest/utils.pm

```diff
@@ -429,7 +429,17 @@ sub wait_guest_online {
     my $guest = shift;
     my $retries = shift // 300;
     # Wait until guest is reachable via ssh
-    script_retry("nmap $guest -PN -p ssh | grep open", delay => 1, retry => $retries);
+    if (script_retry("nmap $guest -PN -p ssh | grep open", delay => 1, retry => $retries, die => 0) != 0) {
+        # Ensure guest is running
+        if (script_run("virsh list --name --state-running | grep $guest") != 0) {
+            script_run("virsh destroy $guest");
+            assert_script_run("virsh start $guest");
+            script_retry("nmap $guest -PN -p ssh | grep open", delay => 1, retry => $retries);
+        }
```

Comment:
I concern this makes too big changes to wait_guest_online(), It doubles the waiting time and restarts the guest if first try fails. I would like to suggest to create a new function as wait_guest_online() is wide used in virt test.

In the case of sriov network card test, there is a test step, attach an vf card to guest then reboot the guest. This function is used to check if guest reboots successfully. In [this failed test](https://openqa.nue.suse.com/tests/8751801#step/sriov_network_card_pci_passthrough/1051), it took 30 minutes and lots of screenshots were printed when guest failed to bootup. but with this change, the test would be aborted after 60 minutes. I don't think it is afordable.
Meanwhile, if a device is hotplugged to the guest in live mode(in contrast to persistent/config mode), virsh destroy + virsh start may not bring up the original guest, it depends on hypervisor.

Reply:
Generally speaking, it does not double waiting time. This change is to check whether the guest is in running after failure. If it is not in running, then it will be started forcibly to be given a second chance to retry.

Reply:
but I found another issue. For example, my test is to see whether the guest can reboot normally after a series of operations. Your retry may cover up the problem.

Reply:
I added $state_check which is 0 by default:
```
        # Ensure guest is running
        if (($state_check != 0) and (script_run("virsh list --name --state-running | grep $guest") != 0)) {
            script_run("virsh destroy $guest");
            assert_script_run("virsh start $guest");
            script_retry("nmap $guest -PN -p ssh | grep open", delay => 1, retry => $retries);
        }
```

Reply:
then that's ok for me.